spec/celebration_spec.rb

```diff
@@ -55,4 +55,13 @@
     it { expect(nc.sanctorale?).to be false }
     it { expect(nc.temporale?).to be false }
   end
+
+  describe '#to_s' do
+    I18n.with_locale(:en) do
+      celebration = CR::PerpetualCalendar.new[Date.new(2000, 1, 8)].celebrations[0]
+      it { expect(celebration.to_s)
+          .to eq "#<CalendariumRomanum::Celebration @title=Saturday after Epiphany @rank=#<CalendariumRomanum::Rank @priority=3.13 desc=#{celebration.rank.desc.inspect}> @colour=#<CalendariumRomanum::Colour white> symbol=nil date=nil cycle=:temporale>"
```

Comment:
- The string value of `@title` should be double-quoted
- The expectation should not rely on the implementation under test, i.e. `celebration.rank.desc` should not be called, but the value written literally.
- There must be no test/setup code in the `describe` block outside of the `it` block. Setup shared by multiple examples can be extracted to a `before` block, but it's not necessary here as there is only one example. Just move everything in the `it` block.
- Everywhere in the project, multi-line blocks are written in the `do ... end` form, so I would prefer it this way here, too (but this is the least important issue)

Reply:
Regarding the use of `#{celebration.rank.desc}`, I used it only because the way it gets rendered when called normally ignores the settings from `I18n.with_locale` block and would cause the check to fail for any locale other than English. Believe me, I spent a good couple hours trying to figure out a better workaround, but short of modifying the Rank class itself I'm stumped for a solution.

Reply:
It's because the `I18n.with_locale` call is outside of the `it`, so it takes effect when the test case is being prepared, but not at the time it is being executed. Moving `I18n.with_locale` into the `it` block should fix this.